quipucords/scanner/openshift/entities.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) 2022 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 3 (GPLv3). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv3
+# along with this software; if not, see
+# https://www.gnu.org/licenses/gpl-3.0.txt.
+#
+"""Entities representing data collected from OpenShift."""
+import json
+from dataclasses import dataclass, field
+from typing import Dict, List
+
+
+@dataclass
+class OCPProject:
+    """Entity representing OpenShift Projects/Namespaces."""
+
+    name: str
+    labels: Dict[str, str]
+    deployments: List["OCPDeployment"] = field(default_factory=list)
+    errors: Dict[str, "OCPError"] = field(default_factory=dict)
+
```

Comment:
This is a really good use of the @dataclass. 
Regarding deployments and errors attributes, those won't be created when the object instance is created, right ?
default_factory flag will point to the data type expected?

Reply:
its a function that should be executed if no value is provided when initializing the class.